pkgs/os-specific/linux/ati-drivers/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, kernel ? null, which
+{ stdenv, fetchurl, kernel ? null, linux, which
```

Comment:
That's what I thought too, but when I was running my build locally, `kernel` *was* `null`.  That seemed weird to me, although I don't see `kernel` being passed in explicitly where this is called (at https://github.com/jerith666/nixpkgs/blob/df9b7c1274551bc1ce0b1e71bb35e4cfbad143ae/pkgs/top-level/all-packages.nix#L11276).  But I also don't understand what args `callPackage` is expected to fill in automatically.

At any rate, here is what I get:

```
$ git diff
diff --git a/pkgs/os-specific/linux/ati-drivers/default.nix b/pkgs/os-specific/linux/ati-drivers/default.nix
index d54b0ab..621d756 100644
--- a/pkgs/os-specific/linux/ati-drivers/default.nix
+++ b/pkgs/os-specific/linux/ati-drivers/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, kernel ? null, linux, which
+{ stdenv, fetchurl, kernel ? null, which
 , xorg, makeWrapper, glibc, patchelf, unzip
 , fontconfig, freetype, mesa # for fgl_glxgears
 , # Whether to build the libraries only (i.e. not the kernel module or
@@ -72,7 +72,7 @@ stdenv.mkDerivation rec {
     ./patches/15.9-mtrr.patch
     ./patches/15.9-preempt.patch
     ./patches/15.9-sep_printf.patch ]
-  ++ optionals ( (builtins.compareVersions linux.version "4.6") >= 0 )
+  ++ optionals ( (builtins.compareVersions kernel.version "4.6") >= 0 )
                [ ./patches/kernel-4.6-get_user_pages.patch
                  ./patches/kernel-4.6-page_cache_release-put_page.patch ];
 
$ nixos-rebuild build -I nixpkgs=/home/matt/git/nixos/nixpkgs
building Nix...
building the system configuration...
error: value is null while a set was expected, at /home/matt/git/nixos/nixpkgs/pkgs/os-specific/linux/ati-drivers/default.nix:75:44
(use ‘--show-trace’ to show detailed location information)
```